### PR TITLE
UpdateSites customly updating data are not able to retrieve the updated version

### PR DESCRIPTION
**Pull request by varyvol**
Automated test that proves a functionality that was lost with #3428.

How to fix the issue is unclear.

### CODE REVIEW

**Review by oleg-nenashev — APPROVED** [review]

File: test/src/test/java/hudson/model/UpdateSiteTest.java [hunk 1]
```diff
@@ -132,6 +133,20 @@ public void shutdownWebserver() throws Exception {
         assertNotNull(us.getPlugin("AdaptivePlugin"));
     }
 
+    @Test
+    public void updateSiteWithCustomUpdateNowUpdatesData() throws IOException {
```
Comment (jglick) [inline test/src/test/java/hudson/model/UpdateSiteTest.java]: Currently fails with

```
org.junit.ComparisonFailure: expected:<1.[2]> but was:<1.[1]>
	at org.junit.Assert.assertEquals(Assert.java:115)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at hudson.model.UpdateSiteTest.updateSiteWithCustomUpdateNowUpdatesData(UpdateSiteTest.java:147)
```

so it should be `@Ignore`d with a note to that effect.

File: test/src/test/resources/hudson/model/custom-update-center-0.json [hunk 2]
```diff
@@ -0,0 +1 @@
...ken-macro","optional":false,"version":"1.5.1"}],"developers":[{"developerId":"lloyd"}],"excerpt":"Sends build notifications to a buddycloud instance.","gav":"org.jenkins-ci.ruby-plugins:buddycloud:0.2.3","labels":["notifier"],"name":"buddycloud","previousTimestamp":"2012-08-04T16:29:44.00Z","previousVersion":"0.2.2","releaseTimestamp":"2012-08-04T16:48:12.00Z","requiredCore":"1.432","scm":"github.com","sha1":"ySjScOdtVhNyxVBUTjdXtRvR1o4=","title":"Buddycloud Plugin","url":"http://updates.jenkins-ci.org/download/plugins/buddycloud/0.2.3/buddycloud.hpi","version":"0.2.3","wiki":"https://wiki.jenkins-ci.org/display/JENKINS/Buddycloud+Plugin"},"bugzilla":{"buildDate":"Nov 04, 2011","dependencies":[],"developers":[{"name":"New Maintainer Needed"},{"developerId":"mdonohue","name":"mdonohue"},{"developerId":"jieryn","email":"jieryn@gmail.com","name":"Jesse Farinacci"}],"excerpt":"This plugin integrates <a href='http://www.bugzilla.org/'>Bugzilla<\/a> into Jenkins.","gav":"org.jvnet.hudson.plugins:bugzilla:1.5","labels":["external"],"name":"bugzilla","previousTimestamp":"2010-01-30T22:33:34.00Z","previousVersion":"1.4","releaseTimestamp":"2011-11-04T15:27:08.00Z","requiredCore":"1.392","sc
```
Comment (jglick) [inline test/src/test/resources/hudson/model/custom-update-center-0.json]: Maybe write a minimal JSON just mentioning `beer`.

**Review by jglick — COMMENTED** [review]
Seems like the clean fix would be for `UpdateSite.updateData` to become `protected`. IIUC the current `UpdateSite` subclass motivating this issue uses Java reflection, which is obviously not something we can support across refactorings.

As a workaround with current core versions, it seems like it would suffice to call `doInvalidateData()`.

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
retracting my approval. Tests need to be fixed before merging

**Comment by varyvol** [comment]
@oleg this didn't pretend to be a fix, but a proof of the failure I had seen.

**Comment by daniel-beck** [comment]
@varyvol Are you planning to address the issue in this PR? Or did you just want to contribute a test?

If the latter, the test should be ignored by default, so it doesn't introduce failures.

**Comment by varyvol** [comment]
@daniel-beck I've ignored the test.

**Pull request closed without merging**